src/test/java/net/minecraftforge/debug/client/rendering/RenderWorldPreTranslucentEventTest.java

```diff
@@ -0,0 +1,82 @@
+package net.minecraftforge.debug.client.rendering;
+
+import java.util.Random;
+
+import com.mojang.blaze3d.matrix.MatrixStack;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.BlockRendererDispatcher;
+import net.minecraft.client.renderer.IRenderTypeBuffer;
+import net.minecraft.client.renderer.IRenderTypeBuffer.Impl;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.client.renderer.RenderTypeLookup;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.texture.OverlayTexture;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+import net.minecraft.world.dimension.DimensionType;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.client.event.RenderWorldPreTranslucentEvent;
+import net.minecraftforge.common.DimensionManager;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+
+/**
+ * A simple mod to test the RenderWorldPreTranslucentEvent. For this test, a
+ * diamond ore "block" is rendered at the coordinates (0, 128, 0).
+ */
+
+@Mod(RenderWorldPreTranslucentEventTest.MODID)
+@Mod.EventBusSubscriber(value = Dist.CLIENT)
+public class RenderWorldPreTranslucentEventTest
+{
+    public static final String MODID = "pre_translucent_render_test";
+    static final boolean ENABLED = false;
+
+    @SubscribeEvent
+    public static void onWorldRenderPreTranslucent(RenderWorldPreTranslucentEvent event)
+    {
+        if(!ENABLED)
+            return;
+
+        // Get instances of the classes required for a block render.
+        MinecraftServer server = Minecraft.getInstance().getIntegratedServer();
+        World world = DimensionManager.getWorld(server, DimensionType.OVERWORLD, false, true);
+        MatrixStack matrixStack = event.getMatrixStack();
+        Impl renderTypeBuffers = event.getRenderTypeBuffers();
+
+        // Get the projected view coordinates.
+        @SuppressWarnings("resource")
+        Vec3d projectedView = Minecraft.getInstance().gameRenderer.getActiveRenderInfo().getProjectedView();
```

Comment:
We have ActiveRenderInfo context where the event is fired, why not just pass it through from there?

Reply:
The event now has direct access to WorldRenderer's ActiveRenderInfo.